README.md

```diff
@@ -390,43 +402,55 @@ public interface ExampleHandler {
 
 ##### Custom-named
 
-It is a sort of combination of `Named` and `Indexed` annotations. You can get needed value by index or by key. Can be multi-level
+It is a sort of combination of `Named` and `Indexed` annotations. You can get
+needed value by index or by key. It also allows getting elements from nested
+objects and arrays, the value will be got in the order the keys and indexes
+are specified. To get a value by key, you should just declare the required key
+like in `@Named` annotation. To get an array by index, you can declare it as
+`[index]`, in such a way you can get values from this array like `{inner index}`.
```

Comment:
'this array'? With such annotation they'll be retrieved by the nth(inner index) key of the object.